### ISSUE TITLE
env variables

### ISSUE THREAD

**Issue by sskender** [instrumentisto/coturn-docker-image#20]
Any plans to support configuration via docker environment variables?

**Comment by tyranron** [comment]
@sskender would you be so kind to explain more detailed what do you mean?

If you want to configure Coturn via env vars, I think there isn't much we can do. The docker image itself is just a thin wrapper around upstream software. So:
- if Coturn itself doesn't support config via env vars, neither we can;
- if Coturn supports it - just use it.

And, of course, we always can do some custom magic via `docker-entrypoint.sh`. But I don't see much sense in it, since Coturn is pretty well configured via CLI args already. And environments such as Docker Compose and Kubernetes allow to map its env vars to CLI args pretty well. So, in practice, we don't feel any friction with it.

**Comment by sskender** [comment]
For larger deployments via docker-compose file it is very convenient for me to use env variables which can be loaded from `.env` file rather than passing cli args (such as realm).

As far as I know, coturn does not support env variables, only cli args and config file. Correct me if I am wrong on that one. That is not a problem as I run this docker image anyway.

So, what I ended up doing was changing  `docker-entrypoint.sh` to read passed env varibles and stick them as cli args.

In a nutshell, my question is, do you think this is a good way to do it? Do you have any other suggestions? Would you like me to polish my solution and create a pull request?

**Comment by tyranron** [comment]
@sskender 

> In a nutshell, my question is, do you think this is a good way to do it?

If it solves you problem and removes friction - yes.

> Do you have any other suggestions?

As I've mentioned, for us the following mapping with `.env` files is quite enough:
```yaml
services:
  coturn:
    container_name: ${COMPOSE_PROJECT_NAME}-coturn
    image: instrumentisto/coturn:4.5
    depends_on: ["coturn-db"]
    command:
      - --log-file=${COTURN_LOG_FILE}
      # ...etc
```
But I realize that for someone it may be not enough.

> Would you like me to polish my solution and create a pull request?

Yes, of course! Somebody else may gain from it! ❤️

**Comment by sskender** [comment]
Hm, I can't say I have seen this kind of mapping before. Are those variables, such as `${COMPOSE_PROJECT_NAME}`, defined in host system or in coturn service in this case? You haven't defined any .env file or environment variables in compose file.

Because what I had in mind, was **passing .env file to docker container** and read them inside, not using any variables from my host system.

I will definitely create a PR when I will have more time on my hands.

**Comment by tyranron** [comment]
@sskender 

> Are those variables, such as ${COMPOSE_PROJECT_NAME}, defined in host system or in coturn service in this case? You haven't defined any .env file or environment variables in compose file.

They are defined in `.env` file along with `docker-compose.yml` and also can be easily overwritten if required like this:
```bash
COMPOSE_PROJECT_NAME=my docker-compose up
```

> I will definitely create a PR when I will have more time on my hands.

👌

**Comment by sskender** [comment]
Thank you very much for your assistance!
